runtime/moonbase/src/lib.rs

```diff
@@ -636,11 +636,27 @@ impl pallet_ethereum::Config for Runtime {
 	type StateRoot = pallet_ethereum::IntermediateStateRoot<Self>;
 }
 
+pub struct EthereumXcmEnsureProxy;
+impl xcm_primitives::EnsureProxy<AccountId> for EthereumXcmEnsureProxy {
+	fn ensure_ok(delegator: AccountId, delegatee: AccountId) -> Result<(), &'static str> {
+		let f = |x: &pallet_proxy::ProxyDefinition<AccountId, ProxyType, BlockNumber>| -> bool {
+			x.delegate == delegatee && (x.proxy_type == ProxyType::Any)
+		};
+		Proxy::proxies(delegator)
+			.0
+			.into_iter()
```

Comment:
Can you just use `iter()` to avoid cloning data? Since you're only searching and returning `()`, `iter()` should be enough.